src/holofuel/pages/CreateOfferRequest/CreateOfferRequest.js

```diff
@@ -174,40 +181,59 @@ export default function CreateOfferRequest ({ history: { push } }) {
         name='notes'
         placeholder='What is this for?'
         ref={register} />
+
       <RecentCounterparties
         styleName='recent-counterparties'
         agents={agents}
         selectedAgentId={counterpartyId}
         selectAgent={selectAgent} />
-      <Button type='submit' wide variant='secondary' styleName='send-button' dataTestId='submit-button'>
-        {modeVerbs[mode]}
-      </Button>
+      <Button type='submit' dataTestId='submit-button' wide variant='secondary' styleName='send-button' disabled={counterpartyId.length === AGENT_ID_LENGTH ? !isCounterpartyFound : true}>Send</Button>
     </form>
   </PrimaryLayout>
 }
 
-export function RenderNickname ({ agentId, setCounterpartyNick }) {
-  const { loading, error, data: { holofuelCounterparty = {} } = {} } = useQuery(HolofuelCounterpartyQuery, {
+export function RenderNickname ({ agentId, setCounterpartyNick, setCounterpartyFound, isCounterpartyFound, newMessage }) {
+  const { loading, error: queryError, data: { holofuelCounterparty = {} } = {} } = useQuery(HolofuelCounterpartyQuery, {
     variables: { agentId }
   })
-  const { nickname } = holofuelCounterparty
+
+  const [hasDisplayedNotFoundMessage, setHasDisplayedNotFoundMessage] = useState(null)
+
+  const { nickname, notFound } = holofuelCounterparty
   useEffect(() => {
     setCounterpartyNick(nickname)
   }, [setCounterpartyNick, nickname])
 
+  useEffect(() => {
+    if (!loading) {
+      if (notFound) {
+        setCounterpartyFound(false)
+        if (!hasDisplayedNotFoundMessage) {
+          newMessage('This HoloFuel Peer is currently unable to be located in the network. \n Please verify the hash, ensure your HoloFuel Peer is online, and try again after a few minutes.')
+          setHasDisplayedNotFoundMessage(true)
+        }
+      } else {
+        setCounterpartyFound(true)
```

Comment:
you need to reset `hasDisplayedNotFoundMessage` here, and in the other else clause.
`setHasDisplayedNotFoundMessage(false)`

Reply:
I though I added these state resets.. not sure when/how it was deleted. :thinking: Nonetheless, good call.